### ISSUE TITLE
Python Jupyter Notebook program output is different from what is shown there

### ISSUE THREAD

**Issue by mayankkaizen** [gedeck/practical-statistics-for-data-scientists#34]
This is in reference to Python Jupyter Notebook for Chapter 5: Classification, section:  Undersampling. 

The codes and outputs are, as mentioned in Notebook, shown below - 

![original](https://user-images.githubusercontent.com/11568906/155891291-b271faf2-59b0-49a9-accc-2725751b7f36.PNG)

However, when I rerun that notebook, the output is as shown below 

![actual](https://user-images.githubusercontent.com/11568906/155891363-4c1e66c0-5015-4117-9c85-11fcd0d475dd.PNG)

Needless to say, the output is drastically different from what is in original notebook. I have rerun the same code in different notebook and yet the output is different from the original. 

Please look into this.

**Comment by gedeck** [comment]
Thank you for reporting. I rebuild an environment and rerun the notebook. As you can see in the screenshot, I also get a different result. However, it is different also from your result. 

<img width="1095" alt="image" src="https://user-images.githubusercontent.com/8720575/155906844-675d442d-92c1-42e4-87b3-22f351e1175c.png">

Logistic regression fits the data using an iterative optimization procedure that minimizes a loss function. Scikit-learn allows using a variety of different solvers; here we use `liblinear`.  Depending on the optimization procedure we can get different results. If I use for example `lbfgs`, the model will predict none of the data points as `default`. Even with `liblinear`, the result changes if you change the value of `tol=1e-4` and if necessary the number of iterations. E.g. by decreasing the tolerance, I get values around 0.3% (which would be closer to the result from the R code). According to the documentation, `liblinear` also has a random component. 
<img width="728" alt="image" src="https://user-images.githubusercontent.com/8720575/155908203-5ef6abd8-ced4-4af5-90d8-edd2ec9da970.png">
However, I could not  see a dependency of the result on the value of `random_state`. 

I looked at the source code for `LogisticRegression` but cannot see any obvious recent changes. The current version which states 0.017 on GitHub is about 6 months old, the previous version was identical to the result that I get now. What caused the intermittent difference, I really cannot say.

**Comment by mayankkaizen** [comment]
Thank you very much for the quick response and clearing the confusion.  

1133.55 vs 19.360 and 0.0166 vs 0.975939. The difference was of orders of magnitude and I was really worried how this can be possible. 

Thank you, again.
